core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -329,7 +329,7 @@ private[deploy] class Worker(
         registrationRetryTimer = Some(forwordMessageScheduler.scheduleAtFixedRate(
           new Runnable {
             override def run(): Unit = Utils.tryLogNonFatalError {
-              self.send(ReregisterWithMaster)
+              Option(self).foreach(_.send(ReregisterWithMaster))
```

Comment:
`self` will return null if RpcEndpoint is stopped (unregistered). Although `registrationRetryTimer` will be cancelled in `onStop`, this `run` method may be running when `onStop` is called.